terraform/provider/resource_teleport_role.go

```diff
@@ -1,61 +1,145 @@
 package provider
 
 import (
-	"github.com/gravitational/teleport-plugins/terraform/tfschema"
+	"context"
 
+	"github.com/gravitational/teleport-plugins/terraform/tfschema"
+	"github.com/gravitational/trace"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/gravitational/teleport/api/types"
+)
+
+const (
+	roleNamePath = "metadata.0.name"
 )
 
+// teleport_role resource definition
```

Comment:
`resourceTeleportRole returns ...` to conform to Godoc style comments.

Reply:
👍